editoast/src/views/timetable/stdcm.rs

```diff
@@ -642,3 +661,89 @@ enum SimulationTimeResult {
     SimulationTime { value: u64 },
     Error { error: Box<SimulationResponse> },
 }
+
+#[cfg(test)]
+mod tests {
+    use pretty_assertions::assert_eq;
+
+    use crate::views::rolling_stock::tests::fast_rolling_stock_form;
+
+    use super::*;
+
+    fn create_test_rolling_stock() -> RollingStockModel {
+        let rs_form = fast_rolling_stock_form("test_rolling_stock");
+        RollingStockModel {
+            id: 5656565,
+            name: rs_form.name,
+            loading_gauge: rs_form.loading_gauge,
+            supported_signaling_systems: rs_form.supported_signaling_systems,
+            version: 33,
+            base_power_class: rs_form.base_power_class,
+            comfort_acceleration: rs_form.comfort_acceleration,
+            inertia_coefficient: rs_form.inertia_coefficient,
+            startup_acceleration: rs_form.startup_acceleration,
+            startup_time: rs_form.startup_time,
+            effort_curves: rs_form.effort_curves,
+            electrical_power_startup_time: rs_form.electrical_power_startup_time,
+            raise_pantograph_time: rs_form.raise_pantograph_time,
+            energy_sources: rs_form.energy_sources,
+            gamma: rs_form.gamma,
+            locked: false,
+            metadata: rs_form.metadata,
+            power_restrictions: rs_form.power_restrictions,
+            railjson_version: "12".to_string(),
+            rolling_resistance: rs_form.rolling_resistance,
+            length: 140.0,       // m
+            mass: 150000.0,      // kg
+            max_speed: 110000.0, // m/s
+        }
+    }
+
+    #[test]
+    fn simulation_with_parameters() {
+        let rolling_stock = create_test_rolling_stock();
+
+        let simulation_parameters = SimulationParameters {
+            total_mass: Some(123.0),
+            total_length: Some(455.0),
+            max_speed: Some(100000.0), // m/s
+        };
+
+        let physics_rolling_stock: PhysicsRollingStock =
+            PhysicsRollingStock::new(rolling_stock.into(), simulation_parameters);
+
+        assert_eq!(physics_rolling_stock.mass, 123_u64);
+        assert_eq!(physics_rolling_stock.length, 455000_u64); // It should be converted in mm
+        assert_eq!(physics_rolling_stock.max_speed, 100000_f64);
+    }
+
+    #[test]
+    fn simulation_without_parameters() {
+        let rolling_stock = create_test_rolling_stock();
+
+        let simulation_parameters = SimulationParameters::default();
+
+        let physics_rolling_stock: PhysicsRollingStock =
+            PhysicsRollingStock::new(rolling_stock.into(), simulation_parameters);
+
+        assert_eq!(physics_rolling_stock.mass, 150000_u64);
+        assert_eq!(physics_rolling_stock.length, 140000_u64); // It should be converted in mm
+        assert_eq!(physics_rolling_stock.max_speed, 110000_f64);
+    }
+
+    #[test]
+    fn max_speed_min_parameters() {
```

Comment:
The naming was a bit confusing to me. Maybe something along the lines of `new_physics_rolling_stock_keeps_the_smallest_available_max_speed` would help understand better the expected behavior that is tested without the need to dive into the code ? At the first read, the value in the `assert_eq` looked like a typo and I had to go into `create_test_rolling_stock` to find the `11000` and understand what the test was about.